src/lib/request/read-many.request.ts

```diff
@@ -0,0 +1,127 @@
+import _ from 'lodash';
+import { FindManyOptions, FindOptionsWhere, LessThan, MoreThan } from 'typeorm';
+
+import { CRUD_POLICY } from '../crud.policy';
+import { Method, PaginationRequest, PaginationResponse, PaginationType, PrimaryKey, Sort } from '../interface';
+import { PaginationHelper } from '../provider';
+
+export class CrudReadManyRequest<T> {
+    private _primaryKeys: PrimaryKey[] = [];
+    private _findOptions: FindManyOptions<T> & { where: FindOptionsWhere<T>; take: number } = {
+        where: {},
+        take: CRUD_POLICY[Method.READ_MANY].default.numberOfTake,
+    };
+    private _sort: Sort;
+    private _pagination: PaginationRequest;
+
+    get primaryKeys() {
+        return this._primaryKeys;
+    }
+    get findOptions() {
+        return this._findOptions;
+    }
+    get pagination() {
+        return this._pagination;
+    }
+
+    setPagination(pagination: PaginationRequest): this {
+        this._pagination = pagination;
+        return this;
+    }
+
+    setPrimaryKey(primaryKeys: PrimaryKey[]): this {
+        this._primaryKeys = primaryKeys;
+        return this;
+    }
+
+    setWithDeleted(withDeleted: boolean): this {
+        this._findOptions.withDeleted = withDeleted;
+        return this;
+    }
+
+    setWhere(where: FindOptionsWhere<T> & Partial<T>): this {
+        this._findOptions.where = where;
+        return this;
+    }
+
+    setTake(take: number): this {
+        this._findOptions.take = take;
+        return this;
+    }
+
+    setSort(sort: Sort): this {
+        this._sort = sort;
+        this._findOptions.order = this._primaryKeys.reduce((order, primaryKey) => ({ ...order, [primaryKey.name]: sort }), {});
+        return this;
+    }
+
+    setRelations(relations: string[] | undefined): this {
+        this._findOptions.relations = relations;
+        return this;
+    }
+
+    generate(): this {
+        if (this.pagination.type === PaginationType.OFFSET) {
+            if (!!this.pagination.limit) {
+                this._findOptions.take = this.pagination.limit;
+            }
```

Comment:
How about using `Boolean()` rather than `!!` in order to show converting types of value?

Reply:
ok thanks :)